src/database/sizeRemote.ts

```diff
@@ -0,0 +1,121 @@
+import * as request from "request";
+import { Response } from "request";
+import * as responseToError from "../responseToError";
+import * as utils from "../utils";
+import * as FirebaseError from "../error";
+import * as logger from "../logger";
+import * as api from "../api";
+
+export interface SizeResult {
+  success: boolean;
+  bytes: number;
+  error: FirebaseError;
+}
+
+export interface SizeRemote {
+  /**
+   * Call the shallow get API with limitToFirst=numSubPath.
+   * @param path the path to list
+   * @param numSubPath the number of subPaths to fetch.
+   * @return the list of sub pathes found.
+   */
+  sizeNode(path: string, timeout: number): Promise<SizeResult>;
+}
+
+export class RTDBSizeRemote implements SizeRemote {
+  private instance: string;
+
+  constructor(instance: string) {
+    this.instance = instance;
+  }
+
+  sizeNode(path: string, timeout: number): Promise<SizeResult> {
+    const url = `${utils.addSubdomain(api.realtimeOrigin, this.instance)}${path}.json`;
+    const params: any = { timeout: `${timeout}ms` };
+
+    const reqOptions = {
+      url,
+    };
+    return api.addRequestHeaders(reqOptions).then((reqOptionsWithToken) => {
+      reqOptionsWithToken.qs = params;
+      return new Promise((resolve, reject) => {
+        let response: any;
+        let errorResponse = "";
+        let payload = "";
+        let erroring = false;
+
+        request
+          .get(reqOptionsWithToken)
+          .on("response", (res) => {
+            response = res;
+            if (response.statusCode >= 400) {
+              erroring = true;
+            }
+          })
+          .on("data", (chunk) => {
```

Comment:
Seems a bit over complicated.
Please check `database-settings-get.ts` for a simpler API.

I don't think we need to use the streaming API.